clang/lib/Gold/GoldElaborator.cpp

```diff
@@ -806,22 +730,86 @@ static clang::Decl *processNamespaceDecl(Elaborator& Elab,
     LastLoc = S->getLoc();
   }
 
-  gold::Scope *NamespaceRep = SemaRef.getCurrentScope();
+  NSDecl->Rep = SemaRef.getCurrentScope();
   SemaRef.getCxxSema().ActOnFinishNamespaceDef(NSDecl, LastLoc);
   SemaRef.leaveClangScope(LastLoc);
   SemaRef.popDecl();
 
-  DeclContext *Owner = SemaRef.getCurrentCxxDeclContext();
-  CppxNamespaceDecl *Wrapper =
-    CppxNamespaceDecl::Create(Context.CxxAST, Owner, D->Decl->getLoc(),
-                              D->getId(), cast<clang::NamespaceDecl>(NSDecl),
-                              NamespaceRep);
-  D->Cxx = Wrapper;
-
   // FIXME: We should be returning a DeclGroupPtr to the NSDecl grouped
   // with the implicit UsingDecl, UD.
   return NSDecl;
 }
+static void handleTemplateParameters(Sema &SemaRef,
+                                     Sema::OptionalScopeRAII &ScopeToInit,
+                                     Sema::OptioanlClangScopeRAII &ClangScope,
+                                     Declaration *D, Declarator *Dcl) {
+  gold::Scope **ScopePtr = nullptr;
+  clang::SourceLocation Loc = Dcl->getLoc();
+  clang::TemplateParameterList *ParamList = nullptr;
+  const Syntax *Params = nullptr;
+  TemplateParamsDeclarator *TPD = Dcl->getAsTemplateParams();
+  ScopePtr = TPD->getScopePtrPtr();
+  Params = TPD->getSyntax();
+
+  // Initializing the scopes we have to deal with.
+  ScopeToInit.Init(SK_Template, Params, ScopePtr);
+  ClangScope.Init(clang::Scope::TemplateParamScope, Loc);
+
+  // Constructing actual parameters.
+  llvm::SmallVector<clang::NamedDecl *, 4> TemplateParamDecls;
+  if (!TPD->isImplicitlyEmpty()) {
```

Comment:
We never include braces on single-line if statements, unless they're followed/preceded by a braced else statement.

There should also be a space between a closing brace and the next line.